paramiko/client.py

```diff
@@ -279,6 +282,12 @@ def connect(
             The targets name in the kerberos database. default: hostname
         :param float banner_timeout: an optional timeout (in seconds) to wait
             for the SSH banner to be presented.
+        :param str pkcs11pin: If using PKCS11, this will be the pin of your
+            token or smartcard.
+        :param str pkcs11provider: If using PKCS11, this will be the provider
+            for the PKCS11 interface. Example: /usr/local/lib/opensc-pkcs11.so.
+        :param str pkcs11session: If using PKCS11 in a multithreaded
+            application you can share the session between threads.
```

Comment:
I second @ploxiln's question - I think what he's getting at is it's unclear what the relationship is between these arguments & the data stored in the session - which, w/o doing a deep analysis, seems to largely be a 3-tuple whose members include the pin and provider anyways.

So if we can simplify the API on our end to "you should have called `pkcs11_open_session()` yourself, please hand me the resulting session obj/tuple" that cleans up a lot of intermediate code in an already-very-crowded space.